src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpWebRequestActivitySource.net461.cs

```diff
@@ -456,96 +445,145 @@ private static void PerformInjection()
-        /// this one time performance hit than to get it multiple times later, or do it lazily and have to worry about
-        /// threading issues. If Initialize has been called before, it will not doing anything.
-        /// </summary>
-        private void Initialize()
+        private void RaiseRequestEvent(HttpWebRequest request)
         {
-            lock (this)
+            if (!WebRequestActivitySource.HasListeners() || this.IsRequestInstrumented(request))
             {
-                if (!this.initialized)
-                {
-                    try
-                    {
-                        // This flag makes sure we only do this once. Even if we failed to initialize in an
-                        // earlier time, we should not retry because this initialization is not cheap and
-                        // the likelihood it will succeed the second time is very small.
-                        this.initialized = true;
+                // No subscribers to the ActivitySource or this request was instrumented by previous
+                // RaiseRequestEvent, such is the case with redirect responses where the same request is sent again.
+                return;
+            }
 
-                        PrepareReflectionObjects();
-                        PerformInjection();
-                    }
-                    catch (Exception ex)
-                    {
-                        // If anything went wrong, just no-op. Write an event so at least we can find out.
-                        this.Write(InitializationFailed, new { Exception = ex });
-                    }
-                }
+            var activity = WebRequestActivitySource.StartActivity(
+                ActivityName,
+                ActivityKind.Client,
+                Activity.Current?.Context ?? default,
+                this.BuildRequestTags(request));
+
+            if (activity == null)
+            {
+                return;
             }
+
+            IAsyncResult asyncContext = readAResultAccessor(request);
+            if (asyncContext != null)
+            {
+                // Flow here is for [Begin]GetResponse[Async] without a prior call to [Begin]GetRequestStream[Async].
+
+                AsyncCallbackWrapper callback = new AsyncCallbackWrapper(request, activity, asyncCallbackAccessor(asyncContext));
+                asyncCallbackModifier(asyncContext, callback.AsyncCallback);
+            }
+            else
+            {
+                // Flow here is for [Begin]GetRequestStream[Async].
+
+                asyncContext = writeAResultAccessor(request);
+                AsyncCallbackWrapper callback = new AsyncCallbackWrapper(request, activity, asyncCallbackAccessor(asyncContext));
+                asyncCallbackModifier(asyncContext, callback.AsyncCallback);
+            }
+
+            InstrumentRequest(request, activity);
+
+            activity.SetCustomProperty("request", request);
```

Comment:
>request [](start = 40, length = 7)

I would recommend to name the property name in more specific way to avoid collision for anyone else using the Activity with other scenarios. 

maybe you can name something like HttpWebRequest, or _HttpWebRequest_,. just to be unlikely wanyone else will use that name. 

This should apply to the response and exceptions too.

Reply:
I updated so the custom properties are named "HttpWebRequest.Request", "HttpWebRequest.Response", and "HttpWebRequest.Exception." The idea being they now have a kind of namespace so they don't collide with what the user might want to do.